src/Microsoft.AspNetCore.SystemWebAdapters.CoreServices/HttpRuntimeFactory.cs

```diff
@@ -2,36 +2,51 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System;
+using System.Web.Caching;
 using Microsoft.AspNetCore.SystemWebAdapters;
 
 namespace Microsoft.Extensions.DependencyInjection;
 
 internal static class HttpRuntimeFactory
 {
-    public static IHttpRuntime Create()
+    public static IHttpRuntime Create(IServiceProvider serviceProvider)
     {
         if (NativeMethods.IsAspNetCoreModuleLoaded())
         {
             var config = NativeMethods.HttpGetApplicationProperties();
-
-            return new IISHttpRuntime(config);
+            return new IISHttpRuntime(config, serviceProvider);
         }
 
-        return new DefaultHttpRuntime();
+        return new DefaultHttpRuntime(serviceProvider);
+    }
+
+    internal abstract class BaseHttpRuntime 
+    {
+        private readonly IServiceProvider serviceProvider;
+
+        protected BaseHttpRuntime(IServiceProvider serviceProvider)
+        {
+            this.serviceProvider = serviceProvider;
+        }
+        public Cache Cache => serviceProvider.GetRequiredService<Cache>();
```

Comment:
```suggestion
        protected BaseHttpRuntime(IServiceProvider serviceProvider)
        {
            Cache = serviceProvider.GetRequiredService<Cache>();
        }

        public Cache Cache { get; }
```

Reply:
I'm fine with caching the cache ( :) ), but it's still being accessed via the DI system initially